src/ray/core_worker/core_worker.cc

```diff
@@ -1273,6 +1338,13 @@ bool CoreWorker::AddActorHandle(std::unique_ptr<ActorHandle> actor_handle,
   reference_counter_->AddLocalReference(actor_creation_return_id, CurrentCallSite());
   direct_actor_submitter_->AddActorQueueIfNotExists(actor_id);
 
+  if (!actor_handle->IsPersistedToGCS()) {
```

Comment:
We should do this after the insertion, in case we already have a handle to this actor that's been resolved.

Reply:
That totally makes sense. I will fix it.

Reply:
Moved to inside if (inserted)